cli/types/index.d.ts

```diff
@@ -174,757 +182,739 @@ declare namespace Cypress {
     }
   }
 
-  interface Chainable {
+  interface Chainable<Subject = any> {
     /**
+     * Create an assertion. Assertions are automatically retried until they pass or time out.
+     *
+     * @alias should
      * @see https://on.cypress.io/and
      */
-    and(chainers: string, value?: any): Chainable
-    and(chainers: string, method: string, value: any): Chainable
-    and(fn: (currentSubject?: any) => void): Chainable
+    and: Chainer<Subject>
 
     /**
+     * Assign an alias for later use. Reference the alias later within a
+     * [cy.get()](https://on.cypress.io/get) or
+     * [cy.wait()](https://on.cypress.io/wait) command with a `@` prefix.
+     *
      * @see https://on.cypress.io/as
      */
-    as(alias: string): Chainable
+    as(alias: string): Chainable<Subject>
 
     /**
+     * Blur a focused element. This element must currently be in focus. If you want to ensure an element is focused before blurring, try using .focus() before .blur().
+     *
      * @see https://on.cypress.io/blur
      */
-    blur(options?: BlurOptions): Chainable
+    blur(options?: Partial<BlurOptions>): Chainable<Subject>
 
     /**
+     * Check checkbox(es) or radio(s). This element must be an `<input>` with type `checkbox` or `radio`.
+     *
      * @see https://on.cypress.io/check
      */
-    check(options?: CheckOptions): Chainable
-    check(value: string | string[]): Chainable // no options
+    check(options?: Partial<CheckOptions>): Chainable<Subject>
+    check(value: string | string[], options?: Partial<CheckOptions>): Chainable<Subject>
 
     /**
+     * Get the children of each DOM element within a set of DOM elements.
+     *
      * @see https://on.cypress.io/children
      */
-    children(options?: LoggableTimeoutable): Chainable
-    children(selector: string, options?: LoggableTimeoutable): Chainable
+    children<E extends Node = HTMLElement>(options?: Partial<Loggable & Timeoutable>): Chainable<JQuery<E>>
+    children<K extends keyof HTMLElementTagNameMap>(selector: K, options?: Partial<Loggable & Timeoutable>): Chainable<JQuery<HTMLElementTagNameMap[K]>>    

```

Comment:
This line allows for tag-based selectors to automatically infer the right element. For example:

```ts
cy.get('form')
.then(el => {
  el // $ExpectType JQuery<HTMLFormElement>
})
.children('input')
.then(el => {
  el // $ExpectType JQuery<HTMLInputElement>
})
```

Of course these can be overridden:
```ts
cy.get<HTMLFormElement>('.main-form')
.then(el => {
  el // $ExpectType JQuery<HTMLFormElement>
})
.children<HTMLInputElement>('.input')
.then(el => {
  el // $ExpectType JQuery<HTMLInputElement>
})
```